packages/hunspell.rb

```diff
@@ -12,27 +13,24 @@ class Hunspell < Package
   is_fake
 
   if ARGV[0] == 'install'
-    puts
-    puts 'Enter your preferred language:'
-    puts '1 = American English'
-    puts '2 = Français'
-    puts '3 = Español'
-    puts '0 = Cancel'
-
-    while version = $stdin.gets.chomp
-      case version
-      when '1'
-        depends_on 'hunspell_en_us'
-        break
-      when '2'
-        depends_on 'hunspell_fr_fr'
-        break
-      when '3'
-        depends_on 'hunspell_es_any'
-        break
-      when '0'
-        abort
+    begin
+      @timeout = 10
+      @user_input = Timeout.timeout(@timeout) do
+        puts
+        puts 'Enter your preferred language:'
+        puts "(The default, American English, will be selected in #{@timeout} seconds if there is no input.)"
+        puts '1 = American English'
+        puts '2 = Français'
+        puts '3 = Español'
+        puts '0 = Cancel'
+        $stdin.gets.chomp
       end
+    rescue Timeout::Error
+      @user_input = 1
     end
+    depends_on 'hunspell_en_us' if @user_input == 1
+    depends_on 'hunspell_fr_fr' if @user_input == 2
```

Comment:
Better to use `case`?

Reply:
case causes issues with `depends_on`.

Reply:
That sounds weird...

Reply:
We used case initially and it worked fine.

Reply:
I'll make it easy for you.
```
      case @user_input
      when '1'
        depends_on 'hunspell_en_us'
        break
      when '2'
        depends_on 'hunspell_fr_fr'
        break
      when '3'
        depends_on 'hunspell_es_any'
        break
      when '0'
        abort
```

Reply:
More specifically, it was breaking inside the begin statement while I was testing that.

You're right, I should test this now that logic is outside that again.

Reply:
Here is what I see @uberhacker:
```
  if ARGV[0] == 'install'
    begin
      @timeout = 10
      @user_input = Timeout.timeout(@timeout) do
        puts
        puts 'Enter your preferred language:'
        puts "(The default, American English, will be selected in #{@timeout} seconds if there is no input.)"
        puts '1 = American English'
        puts '2 = Français'
        puts '3 = Español'
        puts '0 = Cancel'
        $stdin.getc.chomp.to_i
      end
    rescue Timeout::Error
      @user_input = 1
    end
    case @user_input
    when '1'
      depends_on 'hunspell_en_us'
      break
    when '2'
      depends_on 'hunspell_fr_fr'
      break
    when '3'
      depends_on 'hunspell_es_any'
      break
    when '0'
      abort
    end
  end
end
```

```
SyntaxError: /usr/local/lib/crew/packages/hunspell.rb:34: Can't escape from eval with break
/usr/local/bin/crew:171:in `block in print_current_package': undefined method `name' for nil:NilClass (NoMethodError)

  status = 'installed' if @device[:installed_packages].any? { |elem| elem[:name] == @pkg.name }
                                                                                        ^^^^^
        from /usr/local/bin/crew:171:in `any?'
        from /usr/local/bin/crew:171:in `print_current_package'
        from /usr/local/bin/crew:1752:in `block in install_command'
        from /usr/local/bin/crew:1749:in `each'
        from /usr/local/bin/crew:1749:in `install_command'
```

Reply:
(That's the same issue I was seeing before, which I why I moved away from using `case`.

Reply:
Maybe remove the quotes on the numbers since they are integers?

Reply:
That doesn't help. The issue is with the `break`s in there.
```
SyntaxError: /usr/local/lib/crew/packages/hunspell.rb:34: Can't escape from eval with break
```